Copy-Edit Getting Started Guide

jnozzi opened the pull request:
* Edited wording for clarity and readability.
* Grammar corrections.
* Fixed errors in and updated code examples.
* Added image caption for vision impaired.

DivineDominion commented on Docs/Getting Started Guide.md:
```diff
@@ -152,28 +140,29 @@ struct SetOAuthURL: Action {
+Reducers are the **only place** in which you should **modify** application state. Reducers take the current application state and an action then return the new transformed application state. We recommend to provide many small reducers that each handle a subset of your application state.
 
-You can do this implementing a top-level reducer that conforms to the `Reducer` protocol. This reducer will then call individual functions for each different part of the app state.
+You can do this by implementing a top-level reducer that conforms to the `Reducer` protocol. This reducer will then call individual reducer functions for each different part of the app state, according to their areas of responsibility.
```
Excellent changes so far! While we're at it, there's no protocol anymore, just a typealias for the function signature. You may want to change the wording accordingly.

DivineDominion commented on Docs/Getting Started Guide.md:
````diff
@@ -152,28 +140,29 @@ struct SetOAuthURL: Action {
 
 ```swift
-func authenticationReducer(state: AuthenticationState?, action: Action) -> AuthenticationState {
+func authenticationReducer(_ state: AuthenticationState?, _ action: Action) -> AuthenticationState {
````
To match the call above (Line 152), this would have to be `authenticationReducer(_ action: Action, _ state: AuthenticationState)`. I prefer named parameter, to be frank, though.

DivineDominion commented on Docs/Getting Started Guide.md:
```diff
@@ -81,7 +77,7 @@ struct AppState: StateType {
 struct AddAction: Action { }
 
-func appReducer(action: Action, state: AppState?) -> AppState {
+func appReducer(_ action: Action, _ state: AppState?) -> AppState {
```
Line 150 is `appReducer(action: Action, state: State?)` with named parameters, making this inconsistent. I personally prefer named parameters, especially since the function name isn't revealing what the 2 parameters would be, unlike, say `add(x, y)`.

DivineDominion commented on Docs/Getting Started Guide.md:
````diff
@@ -279,10 +270,10 @@ store.subscribe(self) { subcription in
 
 ```swift
-func fetchGitHubRepositories(state: State, store: Store<State>) -> Action? {
+func fetchGitHubRepositories(_ state: State, _ store: Store<State>) -> Action? {
````
Aside from the reducer signatures, what's your motivation to remove parameter names here?

DivineDominion commented on Docs/Getting Started Guide.md:
````diff
@@ -320,24 +311,36 @@ func fetchGitHubRepositories(state: State, store: Store<State>) -> Action? {
+
+```swift
+enum GitHubFetchRequestState {
+    case Loading
````
To comply with Swift naming conventions, I suggest lowercasing the cases to `case loading`, `case repositories`, `case networkError`.

DivineDominion commented on Docs/Getting Started Guide.md:
````diff
@@ -320,24 +311,36 @@ func fetchGitHubRepositories(state: State, store: Store<State>) -> Action? {
+
 ```swift
-func doubleValueIfSmall(state: TestAppState, store: Store<TestAppState>) -> Action? {
+func doubleValueIfSmall(_ state: TestAppState, _ store: Store<TestAppState>) -> Action? {
````
Same as above -- even though the typealias cannot provide parameter names anymore, the implementation may, and thus add context on call sites.

jnozzi commented:
I’m confused about the method signatures. When I found the initial error of swapped arguments in the last two appReducer calls to sub-reducers, I looked up the signature in the repository and confirmed the correct order and also noticed they were unnamed (_), so I reflected that in the corrections. Was I looking in the wrong place?

DivineDominion commented:
Ah, I see: it's just that Swift (4?) enforces name-less signatures when you define blocks and typealiases. That's why you have to write `(_ foo: Foo) -> Bar` both for typealias definitions and block parameters in functions. This essentially conveys: "I don't care about the parameter name, only about the type". That's the intent of the change in Swift, I think; _But_ you can pass in functions with fully qualified parameter names as long as the order of parameter types is the same, so `func baz(aName: Foo) -> Bar` will satisfy `(Foo) -> Bar` and `(_ foo: Foo) -> Bar`. So you write functions as Swift-y as before.

jnozzi commented:
Alright! Sorry for the delay (I got a bit busy). I've amended the PR to include the requested changes. Without specific guidance on the signature matching recommendation portion, I went ahead and added a note explaining the discrepancy and changed "matches" to "resembles". I also changed the protocol references for Reducer to reflect the current state of the repo. I did see other protocol mentions are still protocols in code, so I left those alone.

jnozzi commented:
Any updates on review? Any more changes needed?

mjarvis commented:
Thanks @jnozzi ! Like all contributors I'll invite you to the ReSwift org. That will make future contributions easier in case you are interested! 🎉

jnozzi commented:
Branch deleted and org invite accepted, thanks!

DivineDominion commented:
Sorry for the silence last workweek! Good job 👏
